src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,54 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.RegistrationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_LENGTH_PASSWORD = 6;
+    private static final int MIN_LENGTH_LOGIN = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new RegistrationException("User can't be null");
+        }
+        loginCheck(user);
+        passwordCheck(user);
+        ageCheck(user);
```

Comment:
bad method naming - should be verbs, not nouns

```suggestion
        checkLogin(user);
        checkPassword(user);
        checkAge(user);
```